### PR TITLE
feat: support M1 native mac build

### PR DESCRIPTION
**Pull request by blimmer**
Issue: #837

### Requirements

- [x]  issue was opened to discuss proposed changes before starting implementation. It is important do discuss changes before implementing them (Why should we add it? How should it work? How should it look? Where will it be? ...).
- [x]  during development, `node` version specified in `package.json` was used (ie using [nvm](https://github.com/creationix/nvm)).
- [x]  package versions and package-lock.json were not changed (`npm install --no-save`).
- [x]  app version number was not changed.
- [x]  all new code has tests to ensure against regressions. (N/A - GitHub actions and README only)
- [x] `npm run lint` reports no offenses.
- [x] `npm run test` is error-free.
- [x]  README and CHANGELOG were updated accordingly.
- [x]  after PR is approved, all commits in it are [squashed](https://gitbetter.substack.com/p/how-to-squash-git-commits)

### Description of the Change

This PR updates the MacOS build to build both x64 and the arm64 (M1) version of the app. As mentioned in the linked issue, https://github.com/hovancik/stretchly/issues/837, the process for allowing unsigned MacOS applications on Apple Silicon has changed. Instead of just "right-clicking" and allowing the application to run, you now have to remove the quarantine attribute via the command line. This isn't a great UX, but it's the tradeoff unsigned applications have to make with Big Sur + Apple Silicon macs. I've documented this change in the README.

### Verification Process

I verified that this works on my fork. The release 1.9.0-beta.1 has the M1 version attached: https://github.com/blimmer/stretchly/releases. More details on this are available in the issue: https://github.com/hovancik/stretchly/issues/837.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ jobs:
         node-version: '16.9.1'
     - run: npm install npm -g
     - run: npm install
-    - run: node_modules/.bin/electron-builder build
+    - name: Build MacOS Intel
+      run: node_modules/.bin/electron-builder build -m --x64
+      env:
+        GH_TOKEN: ${{ secrets.github_token }}
+    - name: Build MacOS M1/ARM
+      run: node_modules/.bin/electron-builder build -m --arm64
```
Comment (blimmer) [inline .github/workflows/build.yml]: I needed to explicitly specify `--arm64` to force it to build the M1 version. The GitHub runners are x64/intel.

I'm not sure if there's a "better" way to do this with `electron-builder`.

File: README.md [hunk 2]
```diff
@@ -32,6 +32,19 @@ You can also install *Stretchly* with [Homebrew](https://brew.sh/) by running `b
 
 If you're using [Alfred](https://www.alfredapp.com) on macOS you can use this [Alfred Workflow](https://github.com/KingOfSpades/stretchFred) to interact with Stretchly.
 
+#### macOS Apple Silicon / M1
+
+*Stretchly* is also available as a native M1 application. Currently, it's not published to Homebrew, so you need to download
```
Comment (blimmer) [inline README.md:37]: I didn't know how you have your Homebrew cask configured, so I added this comment. Perhaps `electron-builder` automatically publishes both versions for you? I'm not sure.

Comment (hovancik) [reply README.md:37]: Homebrew version is added by community. Looks like they support m1 as well, but cask would need to be updated, one new Stretchly with m1 is out; https://github.com/Homebrew/homebrew-cask/commit/64a4f75db137589efcd5723ff608b99cb3160983

File: .github/workflows/build.yml (hunk 1 above)
Comment (hovancik) [reply .github/workflows/build.yml]: You should be able to put both arguments together in `package.json`, here's example from documentation:

```
"build": {
  "win": {
    "target": [
      {
        "target": "nsis",
        "arch": [
          "x64",
          "ia32"
        ]
      }
    ]
  }
}
```

File: README.md (hunk 2 above)
Comment (blimmer) [reply README.md:37]: Sounds good - I can have a look at this after merge / first ARM-supported release.

File: .github/workflows/build.yml (hunk 1 above)
Comment (blimmer) [reply .github/workflows/build.yml]: Thanks for the pointer - fixed https://github.com/hovancik/stretchly/pull/1099/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Review by hovancik — APPROVED** [review]

**Comment by flowfire** [comment]
Does the github workflow support the m1 build?

**Comment by blimmer** [comment]
Hey @flowfire , at the time this PR was merged GitHub actions did support building the M1 version.

**Pull request merged**